go-controller/pkg/clustermanager/egressip_controller.go

```diff
@@ -802,17 +809,9 @@ func (eIPC *egressIPClusterController) deleteEgressNode(nodeName string) error {
 }
 
 func (eIPC *egressIPClusterController) initEgressIPAllocator(node *v1.Node) (err error) {
-	var parsedEgressIPConfig *util.ParsedNodeEgressIPConfiguration
-	if util.PlatformTypeIsEgressIPCloudProvider() {
-		parsedEgressIPConfig, err = util.ParseCloudEgressIPConfig(node)
-		if err != nil {
-			return fmt.Errorf("unable to use node %s for egress assignment, err: %v", node.Name, err)
-		}
-	} else {
-		parsedEgressIPConfig, err = util.ParseNodePrimaryIfAddr(node)
-		if err != nil {
-			return fmt.Errorf("unable to use node for egress assignment, err: %v", err)
-		}
+	parsedEgressIPConfig, err := util.GetNodeEIPConfig(node)
+	if err != nil {
+		return fmt.Errorf("failed to get egress IP config for node %s: %v", node.Name, err)
```

Comment:
nit: whenever possible, wrap the error in `%w` (instead of `%v`) in fmt.Errorf, so caller can use IsError() to check it this is an error it cares about.